proxy/grpcproxy/cluster.go

```diff
@@ -47,6 +126,10 @@ func (cp *clusterProxy) MemberUpdate(ctx context.Context, r *pb.MemberUpdateRequ
 }
 
 func (cp *clusterProxy) MemberList(ctx context.Context, r *pb.MemberListRequest) (*pb.MemberListResponse, error) {
+	if cp.wa != nil {
```

Comment:
I think the proxy should be able to hand back (optionally?) its client address as a member list without forcing it to `register` to etcd.